router/pkg/config/config.go

```diff
@@ -577,6 +577,19 @@ type WebSocketConfiguration struct {
 	ForwardInitialPayload bool `yaml:"forward_initial_payload" envDefault:"true" env:"WEBSOCKETS_FORWARD_INITIAL_PAYLOAD"`
 	// Authentication configuration for the WebSocket Connection
 	Authentication WebSocketAuthenticationConfiguration `yaml:"authentication,omitempty"`
+	// SetClientInfoFromInitialPayload configuration for the WebSocket Connection
+	SetClientInfoFromInitialPayload WebSocketSetClientInfoFromInitialPayloadConfiguration `yaml:"set_client_info_from_initial_payload"`
+}
+
+type WebSocketSetClientInfoFromInitialPayloadConfiguration struct {
+	// Enabled true if the Router should set the client info from the initial payload of a Subscription Request to the Subgraph
+	Enabled bool `yaml:"enabled" envDefault:"true" env:"WEBSOCKETS_SET_CLIENT_INFO_FROM_INITIAL_PAYLOAD_ENABLED"`
```

Comment:
I'd recommend `WEBSOCKETS_CLIENT_INFO_FROM_INITIAL_PAYLOAD_ENABLED` rather than using verbs in the naming.